asset-transfer-basic/chaincode-java/src/main/java/org/hyperledger/fabric/samples/assetTransfer/AssetTransfer.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.samples.assettransfer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hyperledger.fabric.contract.Context;
+import org.hyperledger.fabric.contract.ContractInterface;
+import org.hyperledger.fabric.contract.annotation.Contact;
+import org.hyperledger.fabric.contract.annotation.Contract;
+import org.hyperledger.fabric.contract.annotation.Default;
+import org.hyperledger.fabric.contract.annotation.Info;
+import org.hyperledger.fabric.contract.annotation.License;
+import org.hyperledger.fabric.contract.annotation.Transaction;
+import org.hyperledger.fabric.shim.ChaincodeException;
+import org.hyperledger.fabric.shim.ChaincodeStub;
+import org.hyperledger.fabric.shim.ledger.KeyValue;
+import org.hyperledger.fabric.shim.ledger.QueryResultsIterator;
+
+import com.owlike.genson.Genson;
+
+/**
+ * Java implementation of the Fabric Car Contract described in the Writing Your
+ * First Application tutorial
+ */
+@Contract(
+        name = "AssetTransfer",
+        info = @Info(
+                title = "Asset Transfer Basic Sample",
+                description = "The hyperlegendary asset transfer",
+                version = "0.0.1-SNAPSHOT",
+                license = @License(
+                        name = "Apache 2.0 License",
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.html"),
+                contact = @Contact(
+                        email = "a.transfer@example.com",
+                        name = "Adrian Transfer",
+                        url = "https://hyperledger.example.com")))
+@Default
+public final class AssetTransfer implements ContractInterface {
+
+    private final Genson genson = new Genson();
+
+    private enum AssetTransferErrors {
+        ASSET_NOT_FOUND,
+        ASSET_ALREADY_EXISTS
+    }
+
+    /**
+     * Creates some initial Assets on the ledger.
+     *
+     * @param ctx the transaction context
+     */
+    @Transaction()
+    public void InitLedger(final Context ctx) {
+        ChaincodeStub stub = ctx.getStub();
+
+        CreateAsset(ctx, "asset1", "blue", 5, "Tomoko", 300);
+        CreateAsset(ctx, "asset2", "red", 5, "Brad", 400);
+        CreateAsset(ctx, "asset3", "green", 10, "Jin Soo", 500);
+        CreateAsset(ctx, "asset4", "yellow", 10, "Max", 600);
+        CreateAsset(ctx, "asset5", "black", 15, "Adrian", 700);
+        CreateAsset(ctx, "asset6", "white", 15, "Michel", 700);
+
+    }
+
+    /**
+     * Creates a new asset on the ledger.
+     *
+     * @param ctx the transaction context
+     * @param assetID the assetID of the new asset
+     * @param color the color of the new asset
+     * @param size the size for the new asset
+     * @param owner the owner of the new asset
+     * @param appraisedValue the appraisedValue of the new asset
+     * @return the created Asset
+     */
+    @Transaction()
```

Comment:
as a best practice in annotation 
```suggestion
    @Transaction(intent=Transaction.TYPE.SUBMIT)
```

same for Update/Delete TransferAsset transactions